### PR TITLE
Improved Classifier/MixtureClassifier/ExpertMixture

### PR DESCRIPTION
**Pull request by regislebrun**
Now, both grade() and classify() methods are parallelized and/or vectorized.

### CODE REVIEW

**Comment by jschueller** [comment]
It's not consistent with the rest of the code to use a flag to use the parallel implementation ; everywhere else you have to use the global TBB::Disable.

**Comment by dbarbier** [comment]
@jschueller There is no use of TBB::Enable/Disable currently, so for consistency reasons I suggest to not use them here ;-)
We have MaximumLikelihoodFactory-Parallel and Distribution-Parallel resource keys, I would thus prefer to rename Classifier-UseTBB into Classifier-Parallel, and also mimic DistributionImplementation: add an isParallel_ member, and isParallel/setParallel methods.

**Comment by jschueller** [comment]
I agree

File: lib/src/Uncertainty/Algorithm/Classification/MixtureClassifier.cxx [hunk 1]
```diff
@@ -77,6 +78,48 @@ UnsignedInteger MixtureClassifier::classify(const NumericalPoint& inP) const
...
 
+Indices MixtureClassifier::classify(const NumericalSample & inS) const
+{
+  if (inS.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the sample to classify has dimension=" << inS.getDimension() << " but the classifier expects dimension=" << mixture_.getDimension();
+  if (ResourceMap::GetAsBool("Classifier-Parallel"))
+    return classifyParallel(inS);
+  return classifySample(inS);
+}
+
+Indices MixtureClassifier::classifySample(const NumericalSample & inS) const
```
Comment (dbarbier) [inline lib/src/Uncertainty/Algorithm/Classification/MixtureClassifier.cxx]: There is no need to redefine classify, just rename classifySample into classifySequential.

**Comment by jschueller** [comment]
+1

File: lib/src/Base/Algo/openturns/ClassifierImplementation.hxx [hunk 2]
```diff
@@ -51,12 +51,24 @@ public:
...
+public:
+
   /** Grade a point as if it were associated to a class */
   virtual NumericalScalar grade(const NumericalPoint & inP,
                                 const UnsignedInteger outC) const;
   virtual NumericalPoint grade(const NumericalSample & inS,
                                const Indices & outC) const;
 
+protected:
+  virtual NumericalPoint gradeTBB(const NumericalSample & inS,
```
Comment (dbarbier) [inline lib/src/Base/Algo/openturns/ClassifierImplementation.hxx]: I would prefer gradeParallel over gradeTBB, but this is not a big deal, +1 from me

**Comment by sofianehaddad** [comment]
+1, thanks @regislebrun

**Pull request merged**